packages/graphics-extras/src/drawRoundedShape.ts

```diff
@@ -0,0 +1,222 @@
+import { PI_2 } from '@pixi/core';
+
+import type { IPointData } from '@pixi/core';
+import type { Graphics } from '@pixi/graphics';
+
+/**
+ * Typed and cleaned up version of:
+ * https://stackoverflow.com/questions/44855794/html5-canvas-triangle-with-rounded-corners/44856925#44856925
+ * @method PIXI.Graphics#drawRoundedShape
+ * @param {PIXI.Graphics} g - Graphics to be drawn on
+ * @param {(IPointData & {radius?: number})[]} points - Minimum length is 3
+ * @param {number} radius - Corners default radius
+ * @returns {PIXI.Graphics} This Graphics object. Good for chaining method calls
+ */
+function roundedShapeArc(
+    g: Graphics,
+    points: (IPointData & { radius?: number })[],
+    radius: number
+): void
+{
+    const vecFrom = (
+        p: { x: number; y: number },
+        pp: { x: number; y: number }
+    ) =>
+    {
+        const x = pp.x - p.x;
+        const y = pp.y - p.y;
+        const len = Math.sqrt((x * x) + (y * y));
+        const nx = x / len;
+        const ny = y / len;
+        const ang = Math.atan2(ny, nx);
+
+        return { x, y, len, nx, ny, ang };
+    };
+
+    let p1 = points[points.length - 1];
+
+    for (let i = 0; i < points.length; i++)
+    {
+        let p2 = points[i % points.length];
+        const p3 = points[(i + 1) % points.length];
+        const v1 = vecFrom(p2, p1);
+        const v2 = vecFrom(p2, p3);
+        let angle = Math.asin((v1.nx * v2.ny) - (v1.ny * v2.nx));
+        let radDirection = 1;
+        let drawDirection = false;
+
+        if ((v1.nx * v2.nx) - (v1.ny * -v2.ny) < 0)
+        {
+            if (angle < 0)
+            {
+                angle = Math.PI + angle;
+            }
+            else
+            {
+                angle = Math.PI - angle;
+                radDirection = -1;
+                drawDirection = true;
+            }
+        }
+        else if (angle > 0)
+        {
+            radDirection = -1;
+            drawDirection = true;
+        }
+
+        const pRadius = p2.radius || radius;
```

Comment:
```suggestion
        const pRadius = p2.radius ?? radius;
```

Maybe it would be better to use `??` here, for example if someone wants to set `radius: 0` at a point to make it a sharp corner.